wordpress-mobile/WordPress-Android
# Plans on JP: Free domain selection and plan selection

Adds the following to plans card on dashboard
- Free domain selection, and 
- Plan selection in a WebView, and
- Simplified conditions for showing plans card

Fixes: #18420
Fixes: #18421 

| Plans card | Domain Selection | Plan Selection |
|--------|--------|--------|
|![Screenshot_20230516_201148](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/bb8f6239-ac31-41b9-8de5-de6e45732370)|![Screenshot_20230518_093442](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/0bf68376-df92-4d3b-81e2-f51ba26cfbe7)|![Screenshot_20230518_093550](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/41fbaf58-68be-4758-be1c-86876a943934)|


To test:

- Launch Jetpack app
- Go to **App Settings** (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Tap **Debug Settings**
- Find **dashboard_card_free_to_paid_plans** [under Remote features](https://github.com/wordpress-mobile/WordPress-Android/pull/18418)
- Enable the flag, and restart the app (scroll down)
- Verify **plans card** appears as shown above on dashboard
- Tap on **plans card**
- Verify it navigates to **Register domain** screen, and domains show label **Free for the first year**
- Select a domain, and Tap on **Select domain** button
- Verify that it goes to **Plans page** on the web


>**Note**
> - The url and display for plans page could change [based on this Calypso PR](https://github.com/Automattic/wp-calypso/pull/77029)
> - hasMappedDomains [is based on the same function in iOS](https://github.com/wordpress-mobile/WordPress-iOS/blob/b73a00f32b811880262e99b168a58e6a2f229bea/WordPress/Classes/Models/Blog.m#L312)

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and manual tests

3. What automated tests I added (or what prevented me from doing so)
Updated and added unit tests

PR submission checklist:
